File: backend/application-service/src/main/java/com/careerflow/applicationservice/offer/model/Offer.java
package com.careerflow.applicationservice.offer.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@Table(name = "offers")
@Getter
@Setter
@NoArgsConstructor
public class Offer {

    @Id
    private UUID id;

    @Column(name = "application_id", nullable = false, unique = true)
    private UUID applicationId;

    @Column(name = "base_salary")
    private BigDecimal baseSalary;

    @Column(name = "joining_bonus")
    private BigDecimal joiningBonus;

    @Column(name = "annual_bonus")
    private BigDecimal annualBonus;

    @Column(name = "stock_value")
    private BigDecimal stockValue;

    @Column(name = "currency", length = 3)
    private String currency;

    @Column(name = "joining_date")
    private LocalDate joiningDate;

    @Enumerated(EnumType.STRING)
    @Column(name = "offer_status")
    private OfferStatus offerStatus;

    @Column(name = "notes")
    private String notes;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @PrePersist
    void onCreate() {
        if (id == null) {
            id = UUID.randomUUID();
        }
        createdAt = Instant.now();
    }
}

File: frontend/src/types/offer.ts
export type OfferStatus = 'PENDING' | 'ACCEPTED' | 'REJECTED' | 'NEGOTIATING' | 'EXPIRED'

export interface Offer {
  id: string
  applicationId: string
  baseSalary: number | null
  joiningBonus: number | null
  annualBonus: number | null
  stockValue: number | null
  currency: string
  joiningDate: string | null
  offerStatus: OfferStatus | null
  notes: string | null
  createdAt: string
}

export interface UpsertOfferRequest {
  baseSalary?: number | null
  joiningBonus?: number | null
  annualBonus?: number | null
  stockValue?: number | null
  currency: string
  joiningDate?: string | null
  offerStatus?: OfferStatus | null
  notes?: string | null
}
